File: RPS/lab 6/prime.py
import random

MAX_PRIME = 999999937

def is_prime(n):                 # optimized trial division with 6k optimization

    def check_divisors(d):       # check the divisors from 5 to sqrt(n) in increments of 6
        if d * d > n:
            return True
        if n % d == 0 or n % (d + 2) == 0:
            return False
        return check_divisors(d + 6)

    if n <= 1:                   # numbers less than or equal to 1 are not prime
        return False
    if n <= 3:                   # 2 and 3 are prime
        return True
    if n % 2 == 0 or n % 3 == 0: # exclude even numbers and numbers divisible by 3
        return False
    return check_divisors(5)

prime_factorization = lambda num, k = 2: [num] if is_prime(num) else [k] + prime_factorization(num // k, k) if num % k == 0 else prime_factorization(num, k + 1)

get_next = lambda i: get_next(i + 1) if not is_prime(i + 1) else i + 1 # find the next prime number

get_random_prime = lambda: get_next(random.randint(2, MAX_PRIME))

def generate_primes(size): # generating an array of prime numbers

    def get_unique_primes(accumulated_primes):
        if len(accumulated_primes) >= size:
            return accumulated_primes
        
        prime = get_random_prime()
        accumulated_primes.append(prime)
        
        return get_unique_primes(accumulated_primes)

    if size <= 0:
        return []  

    return get_unique_primes([])

File: RPS/eclipse-workspace/eclipse test/src/Prime.java

import java.util.Arrays;
import java.util.Random;

public class Prime {
	
	private static int getNext(int prime) {
		int i = prime + 1;
		
		for(; !isPrime(i); i++) {}
		
		return i;
	}
	
	public static int getRandomPrime() {
		Random r = new Random();
		return getNext(r.nextInt(999999999));
	}
	
	private static boolean isPrime(int n) {
		for (int j = 2; j < Math.sqrt(n); j++) {
			if (n % j == 0) {
				return false;
			}
		}
		
		return true;
	}
	
	public static int[] getRandomArray(int length) {
		int[] arr = new int[length];
		
		for(int i = 0; i < length; i++) {
			arr[i] = getRandomPrime();
		}
		
		Arrays.sort(arr);
		
		return arr;
	}



}